### PR TITLE
Improve package README for @pulumi/terraform

### LINKED ISSUE #379: Include a better README in the npm package
We currently embed the README for the terraform bridge in the README for the `@pulumi/terraform` package.  Ideally we would use a for-purpose provider README instead.

### PR DESCRIPTION
**Pull request by jen20**
Fixes #379.

### CODE REVIEW

File: README.package.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+To use this package, please [install the Pulumi CLI first][pulumicli].
+
+## Installing
+
+Currently, the Terraform Provider is available only for Node.js, and is
+distributed as an `npm` package.
+
+### Node.js (JavaScript/TypeScript)
+k
```
Comment (ellismg) [inline README.package.md]: Is the `k` here intentional?

File: README.package.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+To use from JavaScript or TypeScript in Node.js, install using either `npm`:
+
+    $ npm install @pulumi/terraform
+
+or `yarn`:
+a
```
Comment (ellismg) [inline README.package.md]: is the `a` here intentional?

File: README.package.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+```typescript
+import * as pulumi from "@pulumi/pulumi";
+import * as tf from "@pulumi/terraform";
+
+const config = new pulumi.Config();
+
+const ref = new tf.state.RemoteStateReference("remote", {
+    backendType: "remote",
+    organization: "pulumi",
+    token: config.require("terraformEnterpriseToken"),
```
Comment (ellismg) [inline README.package.md]: Could you use requireSecret here?  That will ensure the value is encrypted in the state file.  I suspect we will make bare requiring a value which was secret a warning at some point and an error at some much later point.

**Comment by jen20** [comment]
Thanks @ellismg, I've addressed all of those typos etc in 1955a03 - I'll merge on green.

**Pull request merged**